guppylang/checker/cfg_checker.py

```diff
@@ -131,7 +131,7 @@ def check_bb(
     if bb == cfg.entry_bb:
         assert len(bb.predecessors) == 0
         for x, use in bb.vars.used.items():
-            if x not in cfg.ass_before[bb] and x not in globals.values:
+            if x not in cfg.ass_before[bb] and x not in globals:
```

Comment:
I can't see any changes in the `cfg` directory, which makes me think that `ass_before` still has values in `DefAssignmentDomain == set[str]` rather than the `DefId` which is indexing `globals`?